admin/test/test_images.py

```diff
@@ -116,3 +167,359 @@ def test_multiple_ami(self):
+        build and uploads the AMI ids to a given S3 bucket.
+        """
+        script = _PublishInstallerImagesMain(
+            working_directory=FilePath(self.mktemp())
+        )
+        reactor = object()
+        options = script._parse_options([])
+        result = perform_sequence(
+            seq=packer_publish_sequence(
+                reactor=reactor,
+                template=u"docker",
+                options=options,
+                working_directory=script.working_directory,
+                source_ami=options["source_ami"],
+                ami_map=PACKER_OUTPUT_US_ALL.output,
+            ),
+            eff=script.main_effect(
+                reactor=reactor, options=options
+            )
+        )
+        self.assertEqual(
+            PACKER_OUTPUT_US_ALL.output,
+            result
+        )
+
+
+class PublishInstallerImagesIntegrationTests(TestCase):
+    """
+    Integration test for ``publish-installer-images``.
+    """
+    script = REPOSITORY.descendant(['admin', 'publish-installer-images'])
+
+    def publish_installer_images(self, args, expect_error=False,
+                                 extra_enviroment=None):
+        """
+        Call ``publish-installer-images`` capturing stdout and stderr.
+        """
+        working_directory = FilePath(self.mktemp())
+        working_directory.makedirs()
+        environment = os.environ.copy()
+        if extra_enviroment is not None:
+            environment.update(extra_enviroment)
+        # XXX Don't use TMPDIR because it breaks packer
+        # https://github.com/mitchellh/packer/issues/2792
+        environment["TEMP"] = working_directory.path
+        stdout_path = working_directory.child('stdout')
+        stderr_path = working_directory.child('stderr')
+        self.addDetail(
+            'stdout {!r}'.format(args),
+            content_from_file(
+                stdout_path.path,
+                ContentType('text', 'plain')
+            )
+        )
+        self.addDetail(
+            'stderr {!r}'.format(args),
+            content_from_file(
+                stderr_path.path,
+                ContentType('text', 'plain')
+            )
+        )
+
+        with stdout_path.open('w') as stdout:
+            with stderr_path.open('w') as stderr:
+                try:
+                    return_code = check_call(
+                        [self.script.path] + args,
+                        env=environment,
+                        stdout=stdout, stderr=stderr
+                    )
+                except CalledProcessError as e:
+                    self.addDetail(
+                        'CalledProcessError {!r}'.format(args),
+                        text_content(str(e))
+                    )
+
+                    if expect_error:
+                        return_code = e.returncode
+                    else:
+                        raise
+
+        return (return_code, stdout_path, stderr_path,)
```

Comment:
Yeah, I looked at that but it Eliot logs the stdout and stderr and mixes the two together.